packages/cli/src/commands/rooms.ts

```diff
@@ -0,0 +1,77 @@
+import { flags } from '@oclif/command'
+import { LocationItem, Room } from '@smartthings/core-sdk'
+
+import { APICommand, ListingOutputAPICommand } from '@smartthings/cli-lib'
+
+
+export function buildTableOutput(this: APICommand, data: Room): string {
+	const table = this.newOutputTable()
+	table.push(['Location Id', data.locationId ?? ''])
+	table.push(['Room Id', data.roomId ?? ''])
+	table.push(['Room Name', data.name ?? ''])
+	return table.toString()
+}
+
+export async function getRoomsByLocation(this: APICommand, locationId?: string): Promise<RoomWithLocation[]> {
+	let locations: LocationItem[] = []
+	if (locationId) {
+		this.log(`location specified: ${locationId}`)
+		locations = [await this.client.locations.get(locationId)]
+	} else {
+		locations = await this.client.locations.list()
+	}
+
+	if (!locations || locations.length == 0) {
+		throw Error('could not find any locations for you account. Perhaps ' +
+			"you haven't created any locations yet.")
+	}
+
+	let rooms: RoomWithLocation[] = []
+	for (const location of locations) {
+		const locationRooms = await this.client.rooms.list(location.locationId)
+		rooms = rooms.concat(locationRooms.map(room => { return { ...room, locationName: location.name } }))
+	}
+	return rooms
+}
+
+export type RoomWithLocation = Room & { locationName?: string }
+
+export default class RoomsCommand extends ListingOutputAPICommand<Room, RoomWithLocation> {
+	static description = 'get a specific room'
+
+	static flags = {
+		...ListingOutputAPICommand.flags,
+		locationId: flags.string({
+			char: 'l',
+			description: 'a specific locationId to query',
+		}),
+	}
+
+	static args = [{
+		name: 'idOrIndex',
+		description: 'the room id',
+	}]
+
+	primaryKeyName = 'roomId'
+	sortKeyName = 'name'
+
```

Comment:
We should really be including the locationId in the list output as well. The way you currently do that is to override the `tableHeadings()` method. (You can look at deviceprofiles.ts or capabilities.ts for examples.)

We should also probably support the `--verbose` command line option and include the location name as well when specifying it. Let's write another issue to do that though rather than doing it here so we don't expand the scope of this sprint.